### PR TITLE
[IS-1093] improve db directory name in loopchain

### PR DESCRIPTION
**Pull request by yakkle**
- db directory name exclude `ip:port` when production
- add port to distinguish node when local develop
- rename exact matching old store_path to new store_path

### CODE REVIEW

**Comment by metalg0su** [comment]
Need to check if this PR raises malfunction issues related to DB hot-backup and restoring service (also known as [FASTEST_START](https://github.com/JINWOO-J/prep_docker)).

**Comment by yakkle** [comment]
> Need to check if this PR raises malfunction issues related to DB hot-backup and restoring service (also known as [FASTEST_START](https://github.com/JINWOO-J/prep_docker)).
> 
> All docker images run as `production`?

I discussed with @JINWOO-J about db backup and restore.
This feature requested from infra team and they have agreed to apply next release.
you can check more information in `IS-1093` issue.

File: loopchain/blockchain/blockchain.py [hunk 1]
```diff
@@ -65,10 +65,7 @@ class BlockChain:
     PREPS_KEY = b'preps_key'
     INVOKE_RESULT_BLOCK_HEIGHT_KEY = b'invoke_result_block_height_key'
 
-    def __init__(self, channel_name=None, store_id=None, block_manager=None):
-        if channel_name is None:
-            channel_name = conf.LOOPCHAIN_DEFAULT_CHANNEL
-
+    def __init__(self, channel_name, store_id=None, block_manager=None):
```
Comment (metalg0su) [inline loopchain/blockchain/blockchain.py:68]: Can you remove the default value of `store_id` (required param)?
When I glanced first here, I thought DB name can be `.../db_`.

But I figured out that the variable always be a valid name from what it be initialized.

File: loopchain/utils/loggers/configuration.py [hunk 2]
```diff
@@ -43,7 +43,7 @@ def __init__(self):
         self.log_file_rotate_interval = 0
         self.log_file_rotate_max_bytes = 0
         self.log_file_rotate_backup_count = 0
-        self.log_file_rotate_utf = False
+        self.log_file_rotate_utc = False
```
Comment (metalg0su) [inline loopchain/utils/loggers/configuration.py:46]: 😱

File: loopchain/blockchain/blockchain.py (hunk 1 above)
Comment (yakkle) [reply loopchain/blockchain/blockchain.py]: done

**Review by metalg0su — DISMISSED** [review]

**Review by hyerim-kim — APPROVED** [review]

**Review by metalg0su — APPROVED** [review]

File: loopchain/utils/__init__.py [hunk 3]
```diff
@@ -391,6 +391,34 @@ def parse_target_list(targets: str) -> list:
...
+    Ths method is useless after rename all nodes done.
+
+    :param new_dirname:
+    :return:
+    """
+    storage_path = Path(conf.DEFAULT_STORAGE_PATH)
+    if os.path.exists(storage_path / new_dirname):
+        return
+
+    regex = re.compile(r'[0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3}:[0-9]{4}')
```
Comment (goldworm-icon) [inline loopchain/utils/__init__.py:406]:
```suggestion
    regex = re.compile(r'[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}:[0-9]{4}')
```
or
I recommend to use string splitting and ipaddress package.

**Review by goldworm-icon — CHANGES_REQUESTED** [review]

File: loopchain/channel/channel_service.py [hunk 4]
```diff
@@ -272,13 +272,17 @@ async def __init_peer_auth(self):
...
+        develop = command_arguments.command_values.get(command_arguments.Type.Develop, False)
+        store_id = f"{ChannelProperty().peer_port}_{channel_name}" if develop else channel_name
         try:
             self.__block_manager = BlockManager(
                 channel_service=self,
                 peer_id=ChannelProperty().peer_id,
-                channel_name=ChannelProperty().name,
-                store_identity=ChannelProperty().peer_target
+                channel_name=channel_name,
+                store_identity=store_id
```
[the remark this answers is not in the document]
Comment (yakkle) [reply loopchain/channel/channel_service.py]: done.

File: loopchain/utils/__init__.py (hunk 3 above)
Comment (yakkle) [reply loopchain/utils/__init__.py]: done.

File: loopchain/launcher.py [hunk 5]
```diff
@@ -171,11 +176,10 @@ def start_as_score(args):
...
     additional_conf = {
         "log": {
-            "filePath": f"./log/{network_type}/{channel}/iconservice_{amqp_key}.log"
+            "filePath": f"./log/{network_type}/{db_suffix}/iconservice.log"
         },
-        "iissDbRootPath": conf.DEFAULT_STORAGE_PATH + f"/.iiss_{amqp_key}_{channel}",
-        "scoreRootPath": conf.DEFAULT_STORAGE_PATH + f"/.score_{amqp_key}_{channel}",
-        "stateDbRootPath": conf.DEFAULT_STORAGE_PATH + f"/.statedb_{amqp_key}_{channel}",
+        "scoreRootPath": conf.DEFAULT_STORAGE_PATH + f"/.score_{db_suffix}",
+        "stateDbRootPath": conf.DEFAULT_STORAGE_PATH + f"/.statedb_{db_suffix}",
```
Comment (goldworm-icon) [inline loopchain/launcher.py:182]:
```suggestion
        "scoreRootPath": os.path.join(conf.DEFAULT_STORAGE_PATH, f".score_{db_suffix}"),
        "stateDbRootPath": os.path.join(conf.DEFAULT_STORAGE_PATH, f".statedb_{db_suffix}"),
```

File: loopchain/utils/__init__.py [hunk 6]
```diff
@@ -391,23 +391,56 @@ def parse_target_list(targets: str) -> list:
...
+    Ths method is useless after rename all nodes done.
+
+    :param new_dirname:
+    :return:
+    """
+    storage_path = Path(conf.DEFAULT_STORAGE_PATH)
+    if os.path.exists(storage_path / new_dirname):
+        return
+
+    regex = re.compile(r'([0-9]{1,3}\.){3}[0-9]{1,3}:[0-9]{4}')
```
Comment (goldworm-icon) [inline loopchain/utils/__init__.py:406]: I don't know why you fix the number of port digits to 4.
Some node can use non-default port which is less than 1000 or more than 9999.

**Review by goldworm-icon — CHANGES_REQUESTED** [review]

**Review by windies21 — APPROVED** [review]

File: loopchain/utils/__init__.py (hunk 6 above)
Comment (yakkle) [reply loopchain/utils/__init__.py]: regex removed.

File: loopchain/launcher.py (hunk 5 above)
Comment (yakkle) [reply loopchain/launcher.py:182]: keep current

File: loopchain/blockchain/blockchain.py [hunk 7]
```diff
@@ -78,8 +75,9 @@ def __init__(self, channel_name=None, store_id=None, block_manager=None):
         self.__peer_id = ChannelProperty().peer_id
         self.__block_manager: BlockManager = block_manager
 
-        store_id = f"{store_id}_{channel_name}"
-        self._blockchain_store, self._blockchain_store_path = utils.init_default_key_value_store(store_id)
+        # FIXME : old_store_id is temporary code to rename dbpath
+        old_store_id = f"{ChannelProperty().peer_target}_{channel_name}"
```
Comment (goldworm-icon) [inline loopchain/blockchain/blockchain.py:79]: Just question
What is the difference between ChannelProperty().peer_id and peer_target?
I just worry about the case when peer_id is not the same as peer_target.

File: loopchain/channel/channel_service.py [hunk 8]
```diff
@@ -280,13 +280,17 @@ async def __init_peer_auth(self):
             utils.exit_and_msg(f"peer auth init fail cause : {e}")
 
     def __init_block_manager(self):
-        logging.debug(f"__load_block_manager_each channel({ChannelProperty().name})")
+        logging.debug(f"__init_block_manager() : channel({ChannelProperty().name})")
+
+        channel_name = ChannelProperty().name
+        develop = command_arguments.command_values.get(command_arguments.Type.Develop, False)
+        store_id = f"{ChannelProperty().peer_port}_{channel_name}" if develop else channel_name
```
Comment (goldworm-icon) [inline loopchain/channel/channel_service.py:287]: Just question.
What is store_id?
It seems that the format of old_store_id is different from that of store_id.

Which one is the correct format of store_id?
1. f"{peer_target}"
2. f"{peer_target}_{channel_name}"
3. f"{peer_port}_{channel_name}

Comment (yakkle) [reply loopchain/channel/channel_service.py:287]: It's a just name to distinguish path of key-value store.
The name depends on the case of `develop` or `product`.
It need several paths in `develop`, but one path in `product`.
- `{peer_port}_{channel_name}` in develop
- `{channel_name}` in product

`{peer_target}_{channel_name}` is old one `ip:port` included in dbpath.

**Review by goldworm-icon — APPROVED** [review]

**Review by windies21 — APPROVED** [review]

File: loopchain/utils/__init__.py [hunk 9]
```diff
@@ -391,23 +389,55 @@ def parse_target_list(targets: str) -> list:
...
+    :param old_store_name: old store name
+    :param store_name: new store name
+    :return:
+    """
+    storage_path = Path(conf.DEFAULT_STORAGE_PATH)
+    if os.path.exists(storage_path / store_name):
+        return
+
+    for dirname in os.listdir(storage_path):
+        if dirname == old_store_name:
```
Comment (metalg0su) [inline loopchain/utils/__init__.py:406]: I guess that `rename_db_dir` changes db name:
- from: `./.storage/old_store_name`
- to `./.storage/store_name`.

I'm wondering why you try to iterate all of directories in `storage_path`.
...
```python
        # Iteration removed
        src_path = storage_path / old_store_name
        dst_path = storage_path / store_name
        os.rename(src_path, dst_path)
```

Comment (yakkle) [reply loopchain/utils/__init__.py]: done!

**Review by metalg0su — APPROVED** [review]

**Review by metalg0su — APPROVED** [review]

**Pull request merged**